[22.15.0] libcurl 7.81.0 does not work

According to requirements, libcurl version must be 7.77.0 or higher, but having 7.81.0 version, make install fails.

~~~
-- Checking for module 'libcurl>=7.77.0'
--   Found libcurl, version 7.81.0
~~~

***Error***:



> [ 96%] Building C object openvasd/CMakeFiles/gvm_openvasd_shared.dir/openvasd.c.o
/home/user/source/test/gvm-libs/openvasd/openvasd.c: In function ‘openvasd_send_request’:
/home/user/source/test/gvm-libs/openvasd/openvasd.c:579:7: error: implicit declaration of function ‘curl_easy_header’; did you mean ‘curl_easy_send’? [-Werror=implicit-function-declaration]
  579 |       curl_easy_header (curl, header_name, 0, CURLH_HEADER, -1, &hname);
      |       ^~~~~~~~~~~~~~~~
      |       curl_easy_send
/home/user/source/test/gvm-libs/openvasd/openvasd.c:579:47: error: ‘CURLH_HEADER’ undeclared (first use in this function); did you mean ‘CURLOPT_HEADER’?
  579 |       curl_easy_header (curl, header_name, 0, CURLH_HEADER, -1, &hname);
      |                                               ^~~~~~~~~~~~
      |                                               CURLOPT_HEADER
/home/user/source/test/gvm-libs/openvasd/openvasd.c:579:47: note: each undeclared identifier is reported only once for each function it appears in
/home/user/source/test/gvm-libs/openvasd/openvasd.c:580:41: error: invalid use of undefined type ‘struct curl_header’
  580 |       response->header = g_strdup (hname->value);
      |                                         ^~
cc1: all warnings being treated as errors
make[2]: *** [openvasd/CMakeFiles/gvm_openvasd_shared.dir/build.make:76: openvasd/CMakeFiles/gvm_openvasd_shared.dir/openvasd.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:585: openvasd/CMakeFiles/gvm_openvasd_shared.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

**Operating system:**

Linux OPENVASTEST 6.5.0-44-generic #44~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Jun 18 14:36:16 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.5 LTS"

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -18,7 +18,7 @@ General build environment:
 
 Specific development libraries:
 * libcjson >= 1.7.14 (util)
-* libcurl >= 7.74.0 (openvasd)
+* libcurl >= 7.83.0 (openvasd)
 * libglib >= 2.42 (all)
 * libgio >= 2.42 (util)
 * zlib >= 1.2.8 (util)

--- a/openvasd/CMakeLists.txt
+++ b/openvasd/CMakeLists.txt
@@ -13,7 +13,7 @@ endif (NOT PKG_CONFIG_FOUND)
 ## Dependency checks
 
 pkg_check_modules (GLIB REQUIRED glib-2.0>=2.42)
-pkg_check_modules (CURL REQUIRED libcurl>=7.77.0)
+pkg_check_modules (CURL REQUIRED libcurl>=7.83.0)
 
 # for json parsing we need cJSON
 pkg_check_modules (CJSON REQUIRED libcjson>=1.7.14)

--- a/util/kb.h
+++ b/util/kb.h
@@ -73,7 +73,7 @@ struct kb_item
   {
     char *v_str; /**< Hold an str value for this kb item. */
     int v_int;   /**< Hold an int value for this kb item. */
-  };             /**< Value of this knowledge base item. */
+  }; /**< Value of this knowledge base item. */
 
   size_t len;           /**< Length of string. */
   struct kb_item *next; /**< Next item in list. */

